### PR TITLE
Improved refdocs for Schema 

### PR DESCRIPTION
**Pull request by rlazo**
The improvements include:

- Clarify that `integer` and `float` types are hints to the model, and the returned data could potentially overflow the respective native types.
- Include missing parameters for some of the functions.
- Add examples for `enumerations` and `obj`.

### CODE REVIEW

**Comment by rlazo** [comment]
cc @andrewheard

**Review by emilypgoogle — APPROVED** [review]

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt [hunk 1]
```diff
@@ -20,7 +20,16 @@ public abstract class StringFormat private constructor(internal val value: Strin
...
-/** Represents a schema */
+/**
+ * Definition of a data type.
+ *
+ * These types can be objects, but also primitives and arrays. Represents a select subset of an
+ * [OpenAPI 3.0 schema object](https://spec.openapis.org/oas/v3.0.3#schema).
+ *
+ * **Note:** while optional, including a the `description` field in your `Schema`s is strongly
+ * encouraged. The more information the model has about what it's expected to generate, the better
+ * the results.
```
Comment (rachelsaunders) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt:31]:
```suggestion
 * **Note:** While optional, including a `description` field in your `Schema` is strongly
 * encouraged. The more information the model has about what it's expected to generate, the better
 * the results.
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt [hunk 2]
```diff
@@ -45,10 +59,14 @@ internal constructor(
...
 
     /**
-     * Returns a schema for a 32-bit integer number
+     * Returns a [Schema] for a 32-bit signed integer number.
      *
-     * @param description: The description of what the parameter should contain or represent
-     * @param nullable: Whether null is a valid value for this schema
+     * **Important:** This [Schema] provides a hint to the model that it should generate a 32-bit
+     * integer, but only guarantees that the value will be an integer. Therefore it's *possible*
+     * that decoding it as an `Int` variable, `int` in Java, could overflow.
```
Comment (rachelsaunders) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt:66]:
```suggestion
     * that decoding it as an `Int` variable (or `int` in Java) could overflow.
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt [hunk 3]
```diff
@@ -90,10 +108,15 @@ internal constructor(
       Schema(description = description, nullable = nullable, type = "NUMBER", format = "double")
 
     /**
-     * Returns a schema for a floating point number
+     * Returns a [Schema] for a single-precision floating-point number.
+     *
+     * **Important:** This [Schema] provides a hint to the model that it should generate a
+     * single-precision floating-point number, but only guarantees that the value will be a number.
+     * Therefore it's *possible* that decoding it as a `Float` variable, `float` in Java, could
```
Comment (rachelsaunders) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt:115]:
```suggestion
     * Therefore it's *possible* that decoding it as a `Float` variable (or `float` in Java) could
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt [hunk 4]
```diff
@@ -124,11 +147,24 @@ internal constructor(
...
-     * Returns a schema for a complex object. In a function, it will be returned as a [JSONObject].
+     * Returns a [Schema] for a complex data type.
+     *
+     * This schema instructs the model to produce data of type object, which has keys of type
+     * `String` and values of type [Schema].
      *
-     * @param properties: The map of the object's fields to their schema
-     * @param description: The description of what the parameter should contain or represent
-     * @param nullable: Whether null is a valid value for this schema
+     * **Example:** A `City` could be represented with the following object `Schema`.
```
Comment (rachelsaunders) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt:155]:
```suggestion
     * **Example:** A `city` could be represented with the following object `Schema`.
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt [hunk 5]
```diff
@@ -153,11 +189,11 @@ internal constructor(
...
 
     /**
-     * Returns a schema for an array.
+     * Returns a [Schema] for an array.
      *
-     * @param items: The schema of the elements of this array
-     * @param description: The description of what the parameter should contain or represent
-     * @param nullable: Whether null is a valid value for this schema
+     * @param item The [Schema] of the elements of stored in the array.
+     * @param description An optional description of what the array represent.
```
Comment (rachelsaunders) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt:195]:
```suggestion
     * @param item The [Schema] of the elements of stored in the array.
     * @param description An optional description of what the array represents.
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt [hunk 6]
```diff
@@ -153,11 +189,11 @@ internal constructor(
     }
 
     /**
-     * Returns a schema for an array.
+     * Returns a [Schema] for an array.
      *
-     * @param items: The schema of the elements of this array
-     * @param description: The description of what the parameter should contain or represent
-     * @param nullable: Whether null is a valid value for this schema
+     * @param item The [Schema] of the elements of stored in the array.
```
Comment (rachelsaunders) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt:194]:
```suggestion
     * @param item The [Schema] of the elements stored in the array.
```

**Review by rachelsaunders — CHANGES_REQUESTED** [review]

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt (hunk 5 above)
Comment (rlazo) [reply firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Schema.kt]: Edited manually to match suggestion.

File: firebase-vertexai/CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Unreleased
 * [fixed] Fixed issue where Firebase App Check error tokens were unintentionally missing from the requests. (#6409)
-
+* [fixed] Clarified in kdocs that `Schema.integer` and `Schema.float` only provide hints to the model. (#6420)
```
Comment (rachelsaunders) [inline firebase-vertexai/CHANGELOG.md:3]:
```suggestion
* [fixed] Clarified in the documentation that `Schema.integer` and `Schema.float` only provide hints to the model. (#6420)
```

**Review by rachelsaunders — DISMISSED** [review]

**Review by rachelsaunders — APPROVED** [review]

**Pull request merged**